prisma/schema.prisma

```diff
@@ -1,39 +1,46 @@
 generator client {
-  provider = "prisma-client-js"
+  provider      = "prisma-client-js"
   binaryTargets = ["native", "linux-musl"]
 }
 
 datasource db {
-  provider = "mysql"
-  url      = env("DATABASE_URL")
+  provider          = "mysql"
+  url               = env("DATABASE_URL")
   shadowDatabaseUrl = env("SHADOW_DATABASE_URL")
 }
 
 model PaybuttonAddress {
-  id          Int        @id @default(autoincrement())
-  address     String     @db.VarChar(255)
-  createdAt   DateTime   @default(now())
-  updatedAt   DateTime   @updatedAt
+  id          Int       @id @default(autoincrement())
+  address     String    @db.VarChar(255)
+  createdAt   DateTime  @default(now())
+  updatedAt   DateTime  @updatedAt
   chainId     Int
   paybuttonId Int
   chain       Chain     @relation(fields: [chainId], references: [id], onUpdate: Restrict)
-  paybutton  Paybutton @relation(fields: [paybuttonId], references: [id], onDelete: Cascade, onUpdate: Restrict)
+  paybutton   Paybutton @relation(fields: [paybuttonId], references: [id], onDelete: Cascade, onUpdate: Restrict)
+
+  @@index([chainId], map: "PaybuttonAddress_chainId_fkey")
+  @@index([paybuttonId], map: "PaybuttonAddress_paybuttonId_fkey")
 }
 
 model Paybutton {
-  id                  Int                   @id @default(autoincrement())
-  providerUserId      String?               @db.VarChar(255)
-  createdAt           DateTime              @default(now())
-  updatedAt           DateTime              @updatedAt
-  addresses           PaybuttonAddress[]
+  id             Int                  @id @default(autoincrement())
+  name           String               @db.VarChar(255)
+  uuid           String               @default(dbgenerated("uuid()"))
+  buttonData     String               @db.LongText
+  providerUserId String?              @db.VarChar(255)
+  createdAt      DateTime             @default(now())
+  updatedAt      DateTime             @updatedAt
+  addresses      PaybuttonAddress[]
+
+  @@unique([name, providerUserId], map: "Paybutton_name_providerUserId_unique_constraint")
 }
 
 model Chain {
-  id                  Int                   @id @default(autoincrement())
-  slug                String                @unique @db.VarChar(255)
-  title               String                @db.VarChar(255)
-  createdAt           DateTime              @default(now())
-  updatedAt           DateTime              @updatedAt
-  paybuttonAddresses  PaybuttonAddress[]
-
+  id                 Int                @id @default(autoincrement())
+  slug               String             @unique @db.VarChar(255)
+  title              String             @db.VarChar(255)
+  createdAt          DateTime           @default(now())
+  updatedAt          DateTime           @updatedAt
+  paybuttonAddresses PaybuttonAddress[]
```

Comment:
How's this to be used? It doesn't seem like it belongs here as a property of a chain.

Reply:
a `PaybuttonAddress` has a foreign key to a `chainId`, so this related attribute is automatically generated, it's all the `PaybuttonAddress`es that point to that chain.

Reply:
Wouldn't this get exceptionally large?... it's all paybutton addresses ever? So if we only supported 1 currency, it would effectively just be every address connected to the system that people have entered?

Reply:
Yes. But all this line does is make so that the PrismaClient is able to access this related field, until we explicitly run `chain.paybuttonAddresses` it's not gonna make a difference.

Also, it's not something we can take off because spelling out this relation is prisma syntax, this is what happens if I take it off:
![image](https://user-images.githubusercontent.com/21281174/174446145-fee841ef-0a64-4547-9e56-f2651d0dfaee.png)
There is further info [here](https://www.prisma.io/docs/concepts/components/prisma-schema/relations)

Reply:
Ok I'd like @ScottMcDermid or @Gattaca1 to review this part as well as I don't fully understand the implications.

Reply:
Ya looks like a harmless one-to-many association.

> Wouldn't this get exceptionally large?
Large in which way? Storage or computation? There's no added storage, since all addresses need to belong to some chain. Querying over all addresses in a chain is going to computationally difficult whether or not we're joining on the associated chain.